activerecord/lib/active_record/model_schema.rb

```diff
@@ -447,7 +477,30 @@ def load_schema
         end
 
         def load_schema!
-          @columns_hash = connection.schema_cache.columns_hash(table_name).except(*ignored_columns)
+          @columns_hash = connection.schema_cache.columns_hash(table_name)
+
+          if skip_defining_attributes_from_schema
+            if ignored_columns.any?
+              raise ArgumentError, "can't use ignored_columns and skip_defining_attributes_from_schema at the same time"
```

Comment:
Let's handle this in the writer method for `define_attributes_from_schema` and `ignored_columns`. We want to error early, and we shouldn't assume this is the only place the two can interact.